rtengine/imageio.cc

```diff
@@ -83,6 +85,43 @@ Glib::ustring to_utf8 (const std::string& str)
     }
 }
 
+unsigned short addSat (unsigned short a, unsigned short b)
```

Comment:
I think we should really check the generated assembly here (but only here) with different versions of GCC and then use that knowledge to decide whether saturated addition should get the @heckflosse treatment.
If GCC 4.9 is already properly vectorizing the loop then there is probably no use to it, but I'd still argue that saturated addition would fit well into `rt_math.h` or a similar generic header.